### DISCUSSION TITLE
Negative and Positive Values ONLY (esphome config file suggestion)

### DISCUSSION THREAD

**Discussion by ideal2545** [CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter#59] [Ideas]
Hey there, 

In the US, especially SoCal, power meters 'roll back' when feeding the grid from Solar. So if you have clamps on the two branches feeding your house, the clamps report positive values when getting power from the Grid but then reverse and go negative when you feed energy to the grid. 

HA does not support (currently) any sort of option to specificy negative readings as 'return to grid'. So I adjusted the esphome config offered in the main repository to include simply splitting out negative and positive values to more accurately feed the Energy Panel in HA in this scenario. 

```
#Total House Watts Negative
  - platform: template
    name: ${disp_name} Total House Negative Watts
    id: totalHouseNegativeWatts
    lambda: |-
      if (id(totalHouseWatts).state > 0) {
        return 0;
      } else {
        return abs(id(totalHouseWatts).state) ;
      }
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: ${update_time}

 #Total House Watts Positive
  - platform: template
    name: ${disp_name} Total House Positive Watts
    id: totalHousePositiveWatts
    lambda: |-
      if (id(totalHouseWatts).state < 0) {
        return 0;
      } else {
        return id(totalHouseWatts).state ;
      }
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: ${update_time}

#House Negative kWh
  - platform: total_daily_energy
    name: ${disp_name} House Negative kWh
    power_id: totalHouseNegativeWatts
    filters:
      - multiply: 0.001
    unit_of_measurement: kWh

#House Positive kWh
  - platform: total_daily_energy
    name: ${disp_name} House Positive kWh
    power_id: totalHousePositiveWatts
    filters:
      - multiply: 0.001
    unit_of_measurement: kWh

```

If theres cleaner code or a better way to do this let me know!

Jon

**Comment by profucius** [comment]
Referenced this in a ticket here: https://github.com/CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter/issues/63
